compiler/src/main/java/dagger/internal/codegen/ComponentDescriptor.java

```diff
@@ -410,9 +412,27 @@ private ComponentDescriptor create(
       for (TypeElement componentDependency : componentDependencyTypes) {
         List<ExecutableElement> dependencyMethods =
             ElementFilter.methodsIn(elements.getAllMembers(componentDependency));
+        HashMultimap<QualifiedMethodNameAndType, Equivalence.Wrapper<TypeMirror>> componentMethodType
+                = HashMultimap.create();
         for (ExecutableElement dependencyMethod : dependencyMethods) {
           if (isComponentContributionMethod(elements, dependencyMethod)) {
-            dependencyMethodIndex.put(dependencyMethod, componentDependency);
+            if (dependencyMethod.getParameters().isEmpty()) {
+              // If a component dependency extends from two interfaces that both contain the same named method, then
```

Comment:
I'm not specifically handling covariant return types. If you two dependencyMethods with the same name and distinct return types String and Object, these two dependencyMethods will be treated as distinct. Same as in the past. This seems appropriate to me. But I have a lot to learn about dagger. 

I'm failing to think of a case where covariant return types can be problematic? Can you give me an example of where this could be an issue? I'll write a few tests and be sure to handle these cases.